djhtml/__main__.py

```diff
@@ -1,210 +1,152 @@
+    _info(f"{changed_files} template{s} {have} been reindented.")
+    if unchanged_files:
+        s = "s" if unchanged_files != 1 else ""
+        were = "were" if s else "was"
+        _info(f"{unchanged_files} template{s} {were} already perfect!")
+    if problematic_files:
+        s = "s" if problematic_files != 1 else ""
+        _info(
+            f"{problematic_files} template{s} could not be processed due to an error."
         )
-        if unchanged_files:
-            s = "s" if unchanged_files != 1 else ""
-            were = "were" if s else "was"
-            print(
-                f"{unchanged_files} template{s} {were} already perfect!",
-                file=sys.stderr,
-            )
-        if problematic_files:
-            s = "s" if problematic_files != 1 else ""
-            print(
-                f"{problematic_files} template{s} could not be processed due to an"
-                " error.",
-                file=sys.stderr,
-            )
 
-    sys.exit(changed_files if args.check else problematic_files)
+    # Exit with appropriate exit status
+    if problematic_files:
+        sys.exit(1)
+    if options.check and changed_files:
+        sys.exit(1)
+    sys.exit(0)
 
 
-def _generate_files(input_filenames, suffix):
-    for file_name in input_filenames:
-        if file_name == "-":
-            yield file_name
+def _generate_filenames(paths, suffixes):
+    for filename in paths:
+        if filename == "-":
+            yield filename
         else:
-            file_path = Path(file_name)
-            if file_path.is_file():
-                yield file_path
-            elif file_path.is_dir():
-                yield from _generate_files_from_folder(file_path, suffix)
-
-
-def _generate_files_from_folder(folder, suffix):
-    for file_path in folder.iterdir():
-        if file_path.is_file() and file_path.suffix == suffix:
-            yield file_path
-        elif file_path.is_dir():
-            yield from _generate_files_from_folder(file_path, suffix)
+            path = Path(filename)
+            if path.is_file():
+                yield path
+            elif path.is_dir():
+                yield from _generate_filenames_from_directory(path, suffixes)
+
+
+def _generate_filenames_from_directory(directory, suffixes):
+    for path in directory.iterdir():
+        if path.is_file() and path.suffix in suffixes:
+            yield path
+        elif path.is_dir():
+            yield from _generate_filenames_from_directory(path, suffixes)
+
+
+def _verify_changed(source, result):
+    output_lines = result.split("\n")
+    changed = False
+    for line_nr, line in enumerate(source.split("\n")):
+        if line != output_lines[line_nr]:
+            changed = True
+        if line.strip() != output_lines[line_nr].strip():
+            raise IndentationError("Non-whitespace changes detected. Core dumped.")
+    return changed
+
+
+def _info(msg):
```

Comment:
I messed around with the [logging](https://docs.python.org/3/library/logging.html) module for a while before I concluded it was way to complicated for mere mortals to understand.

Reply:
Would you be open to PRs changing to using the logging module?

Reply:
I don't see the benefits, but if the PR author would convince me of them of course I would be open to them!